crates/circuit/src/circuit_data.rs

```diff
@@ -1528,16 +1539,18 @@ impl CircuitData {
     /// * capacity - The capacity for instructions to use in the output `CircuitData`
     ///     If `None` the length of `other` will be used, if `Some` the integer
     ///     value will be used as the capacity.
-    pub fn clone_empty_like(other: &Self, capacity: Option<usize>) -> Self {
-        CircuitData {
+    pub fn clone_empty_like(py: Python, other: &Self, capacity: Option<usize>) -> PyResult<Self> {
```

Comment:
This change is necessary since `CircuitData::set_global_phase` requires the `Python` token. This function is so far only ever used in `pauli_twirl_2q_gates`, so it is a very non-intrusive change.